src/ui/pages/usersList/types.ts

```diff
@@ -0,0 +1,26 @@
+export enum UserRecipeType { 
+  emailpassword= 'emailpassword', 
+  thirdparty = "thirdparty", 
+  passwordless = "passwordless" }
+
+export type User = { recipeId: UserRecipeType.emailpassword, user: UserEmailPassword } | 
+  { recipeId: UserRecipeType.passwordless, user: UserPasswordless } |
+  { recipeId: UserRecipeType.thirdparty, user: UserThirdParty }
+
+export interface UserEmailPassword {
+  id: string
+  email?: string
+  phoneNumber?: string
+  timeJoined?: number
+  firstName?: string
+  lastName?: string
+}
+
+export interface UserPasswordless extends UserEmailPassword {}
+
+export interface UserThirdParty extends UserEmailPassword {
```

Comment:
Why not just use types for this?

Reply:
Also its strange to see Passwordless inherit from email password, ideally you want

```ts
type User = {...}

type EmailPasswordUser = User & {...}

type PasswordlessUser = User & {...}
```

and so on. You can use type unions and exclusions for addition/removal of properties

Reply:
Its Updated